packages/edit-site/src/components/actions/index.js

```diff
@@ -411,3 +412,77 @@ export const postRevisionsAction = {
 		document.location.href = href;
 	},
 };
+
+export const renamePostAction = {
+	id: 'rename-post',
+	label: __( 'Rename' ),
+	RenderModal: ( { items, closeModal } ) => {
+		const [ item ] = items;
+		const originalTitle = decodeEntities(
+			typeof item.title === 'string' ? item.title : item.title.rendered
+		);
+		const [ title, setTitle ] = useState( () => originalTitle );
+		const { editEntityRecord, saveEditedEntityRecord } =
+			useDispatch( coreStore );
+		const { createSuccessNotice, createErrorNotice } =
+			useDispatch( noticesStore );
+
+		async function onRename( event ) {
+			event.preventDefault();
+			try {
+				await editEntityRecord( 'postType', item.type, item.id, {
+					title,
+				} );
+				// Update state before saving rerenders the list.
+				setTitle( '' );
+				closeModal();
+				// Persist edited entity.
+				await saveEditedEntityRecord( 'postType', item.type, item.id, {
+					throwOnError: true,
+				} );
+				createSuccessNotice( __( 'Name updated' ), {
```

Comment:
This will need to be updated to a bit better message. I was looking at the rest of messages and we are not consistent. Some actions use `posts`, some `pages` and TBH I was thinking of `items`. 

I think for now we could go with `post` or `page`(Post renamed), but it's something we will need decide and update holistically. --cc @jameskoster

Reply:
> This will need to be updated to a bit better message.

I wasn't sure regarding the wording here either but settled on using what was [proposed](https://github.com/WordPress/gutenberg/pull/54648/files#diff-616bcaca326ebbbc7a37c80e07468ea7de74bf2401f6391bda9c980a3f281e55R61) by @SaxonF in https://github.com/WordPress/gutenberg/pull/54648.

We also discussed some of the labelling on Page vs Template actions elsewhere and decided an audit of them all for consistency would be a good follow-up.

Reply:
"$post_type name updated" could be the way to go in the long run, but "Name updated" seems fine in the short term.